sulis_state/src/merchant_state.rs

```diff
@@ -40,19 +41,21 @@ impl MerchantState {
         for item_save in save.items {
             let item = item_save.item;
             let variant = item.variant;
+            let identified = !item.adjectives.contains(&"unidentified".to_string());
             let item = match Module::create_get_item(&item.id, &item.adjectives) {
                 None => invalid_data_error(&format!("No item with ID '{}'", item.id)),
                 Some(item) => Ok(item),
             }?;
 
-            items.add_quantity(item_save.quantity, ItemState::new(item, variant));
+            items.add_quantity(item_save.quantity, ItemState::new(item, variant, identified));
         }
 
         Ok(MerchantState {
             id: save.id,
             loot_list_id: save.loot_list_id,
             buy_frac: save.buy_frac,
             sell_frac: save.sell_frac,
+            identify_fraction: 0.2,
```

Comment:
identify_fraction should not be hardcoded to 0.2.  Merchants are currently defined ad-hoc in conversation files, for example `campaigns/twin_expanse_act1/conversations/dwarf_smith.yml`.

You could `#[serde_default]` this to 0.2 as a default, but it should be changable in the `MerchantData` in `sulis_module/on_trigger`, and it should save in the MerchantSaveState.

Reply:
Fixed